src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputExtension.java

```diff
@@ -0,0 +1,33 @@
+package org.jenkinsci.plugins.workflow.support.steps.input;
+
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+
+public interface InputExtension extends ExtensionPoint {
+    /**
+     * Callback this when an inputStep is executing.
+     * Just for notification, will ignore all exception and log them.
+     * @param inputStep Instance of InputStep
+     * @param run Current building run instance
+     * @param userID Id of a user who triggered the action
+     * @param notifyEvent Event type of notification
+     */
+    void notifyInput(InputStep inputStep, Run run, String userID, NotifyEvent notifyEvent);
```

Comment:
Maybe better to pass `User` instead of `userId` in case someone wants to look at other attributes of the user? Do we really need to pass the whole `InputStep`? I guess if someone wants to examine the set of approvers or something? Maybe also pass a `TaskListener` from the `input` step in case the extension wants to write something to the logs?

Reply:
> Maybe better to pass User instead of userId in case someone wants to look at other attributes of the user?

If someone wants to look at other attributes of the user, they can use this method `getById` from the Class `User`.

> Do we really need to pass the whole InputStep? I guess if someone wants to examine the set of approvers or something? 

Passing the whole InputStep sounds like a little bit heavy. But I cannot find another way to pass the messages of InputStep. In case someone wants to get the parameters and submitter and so on.

> Maybe also pass a TaskListener from the input step in case the extension wants to write something to the logs?

This is a good idea. Accept this and thanks.